.github/workflows/build-and-deploy.yml

```
@@ -0,0 +1,38 @@
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
```

actionlint

missing input "distribution" which is required by action "actions/setup-java@v3". all required inputs are "distribution"

(action)